webapp/src/components/AssetLoader/AssetLoader.container.js

```diff
@@ -0,0 +1,47 @@
+import { connect } from 'react-redux'
+import { push } from 'react-router-redux'
+
+import { locations } from 'locations'
+import { fetchParcelRequest } from 'modules/parcels/actions'
+import { getData as getParcels, getLoading } from 'modules/parcels/selectors'
+import { splitCoordinate } from 'shared/parcel'
+import { getEstates } from 'modules/estates/selectors'
+import { fetchEstateRequest } from 'modules/estates/actions'
+import AssetLoader from './AssetLoader'
+import { ASSET_TYPES } from 'shared/asset'
+
+const mapState = (state, { id, assetType }) => {
+  let assets
+  switch (assetType) {
+    case ASSET_TYPES.parcel:
+      assets = getParcels(state)
+      break
+    case ASSET_TYPES.estate:
+      assets = getEstates(state)
+      break
+  }
+  const isLoading = getLoading(state).some(asset => asset.id === id)
+  const asset = assets[id]
+
+  return {
+    isLoading,
+    asset
+  }
+}
+
+const mapDispatch = (dispatch, { id, assetType }) => {
+  const isParcel = assetType === ASSET_TYPES.parcel
+  const [x, y] = isParcel ? splitCoordinate(id) : [null, null]
```

Comment:
I think this could be `const [x, y] = splitCoordinate(id)` because will return `[0, 0]` if id is not defined and you're already checking for `isParcel` below.

if we want to be extra sure we could add a toString() to splitCoordinate:


```diff
export function splitCoordinate(id) {
  let ids = [0, 0]
  if (id) {
-    ids = id.split(',').map(coord => parseInt(coord, 10))
+    ids = id.toString().split(',').map(coord => parseInt(coord, 10))
  }
  return ids
}
```